sdk/keyvault/azure-keyvault-keys/tests/test_keys_async.py

```diff
@@ -453,11 +453,14 @@ async def test_get_random_bytes(self, client, **kwargs):
 
         generated_random_bytes = []
         for i in range(5):
-            random_bytes = await client.get_random_bytes(count=8)
-            bytes_value = random_bytes.value
-            assert len(bytes_value) == 8
-            assert all([bytes_value != rb] for rb in generated_random_bytes)
-            generated_random_bytes.append(bytes_value)
+            # [START get_random_bytes]
+            # get eight random bytes from a managed HSM
+            result = await client.get_random_bytes(count=8)
+            random_bytes = result.value
+            # [END get_random_bytes]
+            assert len(random_bytes) == 8
+            assert all([random_bytes != rb] for rb in generated_random_bytes)
```

Comment:
`all` isn't doing what you intend because you're giving it an expression that generates lists of bools:
```py
>>> all([s == 's'] for s in ['s'] * 2)
True
>>> all([s != 's'] for s in ['s'] * 2)
True
>>> list([s != 's'] for s in ['s'] * 2)
[[False], [False]]
```
(`[False]` is truthy because it's a non-empty list)

Easy fix--generate bools instead:
```suggestion
            assert all(random_bytes != rb for rb in generated_random_bytes)
```

Reply:
My mistake, thank you for catching this!